### DISCUSSION TITLE
404 and forbidden errors when running with `yarn watch`

### DISCUSSION THREAD

**Discussion by zxytt** [coder/code-server#6200] [Q&A]

**Comment by code-asher** [comment]
Can you provide reproduction steps?  What error do you get?  It seems unlikely this would be caused by code-server.

**Comment by zxytt** [comment]
After the Yarn build is completed, the Yarn watch will start and access the link as shown in
<img width="980" alt="image" src="https://user-images.githubusercontent.com/86701548/236983331-5d046ec3-387c-40a5-9416-579e563a95e5.png">
This package always cannot be built, and occasionally it can. 
![image](https://user-images.githubusercontent.com/86701548/236983422-244e16e0-c312-4569-b4ca-2d0688ca40fe.png)
After building it, accessing the URL will result in the following error
![image](https://user-images.githubusercontent.com/86701548/236988270-2be7d92a-1934-4529-97f1-a9fe621f0fad.png)

**Comment by code-asher** [comment]
Ohhh I see!  The 404 is because VS Code takes much longer to compile than the web server.  You will need to wait for a "Finished compilation" message (there will be two, one for the built-in extensions and another for VS Code itself).  We should detect this and show a better error message though.

The "forbidden" error is because you have not applied the patches to VS Code.  https://github.com/coder/code-server/blob/main/docs/CONTRIBUTING.md#troubleshooting
